### DISCUSSION TITLE
3 encryption jars required

### DISCUSSION THREAD

**Discussion by BilalHaniff** [sassoftware/saspy#445] [Q&A]
My SAS base installation does not include the 3 encryption jars specified below.  
Any idea where I can download these jars from ?
It would be useful if these jars were installed by default.

**ATTN, as of saspy version 3.3.3, the classpath is no longer required in your configuration file!**
The 4 required IOM Java client jars are now included in the saspy repo, and they, along with the saspyiom.jar and the thirdparty CORBA jars will be automatically provided as the calsspath by saspy. This is generally all that is needed. If you require the 3 encryption jars, shown below, they still need to be acquired from your SAS deployment, and then put in the saspy/java/iomclient directory of the saspy install (where the 4 included jars are) so they will be included in the classpath for you.

Note that for AES encryption, at least prior to SAS 9.4M7, Java 8 (release greater than 151), has the needed support in it, so if your workspace server is configured for AES, just having a current version of Java 8 or higher will allow it to work, without needing the 3 encryption jars. That doesn’t work w/ M7 however, so you would need the encryption jars when connecting to M7.

The 3 encryption jars, if needed, can be found in a SAS deployment in a location similar to, but not exactly the sam as, the following. If you get any error about encryption, adding these 3 jars is the proper way to address it.

$SASHome/SASVersionedJarRepository/eclipse/plugins/sas.rutil_904600.0.0.20181017190000_v940m6/sas.rutil.jar
$SASHome/SASVersionedJarRepository/eclipse/plugins/sas.rutil.nls_904600.0.0.20181017190000_v940m6/sas.rutil.nls.jar
$SASHome/SASVersionedJarRepository/eclipse/plugins/sastpj.rutil_6.1.0.0_SAS_20121211183517/sastpj.rutil.jar

**Comment by tomweber-sas** [comment]
What version of SAS do you have installed? Are you connecting to a workspace server that uses one of the encryption methods that require these jars on the client? Yes, it would be great if I could include them in the repo, but that violates laws, so they can't be. If you contact Tech Support, I believe they can provide you these to download.
Tom

**Reply by BilalHaniff** [reply]
SAS 9.4, and I am trying to connect to a workspace server. The SAS installation is a virtual app, and the required jars are therefore not installed in my SASHome folder. I'll have to jump a few hoops and hurdles at my company to find out how to get them installed.

I perfectly understand your reasoning for not adding them

**Reply by tomweber-sas** [reply]
Ok, so you can either copy them over from where your workspace server is deployed; it's a real deployment and will have them. Or, tech support can provide them to you from a download, which sounds like it might be easier than tracking down how to get them from the server. And, just to be clear, there aren't necessarily needed; have you tried connecting and gotten an error indicating they are required? That would be something like number 8 or 9 here: https://sassoftware.github.io/saspy/troubleshooting.html#iom-specific-errors.
Thanks,
Tom

**Reply by BilalHaniff** [reply]
Thanks for the advice. I got them from a colleague, and I'm up and running.

**Reply by tomweber-sas** [reply]
Sweet :)
